riscv-sbi.adoc

```diff
@@ -1674,6 +1675,250 @@ The possible error codes returned in `sbiret.error` are shown in the
+the data blob from `evidence_data_addr`.
+* A https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.1[AuthorityKeyIdentifier]
+X.509 extension.
+
+The possible error codes returned in `sbiret.error` are shown in the
+<<table_get_evidence_errors>> below.
+
+[#table_get_evidence_errors]
+.Attestation Get Evidence Errors
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code          | Description
+| SBI_SUCCESS         | The evidence is generated and the certificate size is
+                        `sbiret.value`.
+| SBI_ERR_FAILED      | The evidence could not be generated.
+| SBI_INVALID_ADDRESS | The CSR or the certificate adress is invalid.
+| SBI_INVALID_PARAM   | The CSR at `csr_addr` is invalid.
+|===
+
+[NOTE]
+.Attestation Flow Example
+====
+Guest userspace flow TBD
+====
+
+[#measurement-extend-function,reftext="Extend Measurement"]
+=== Function: Extend Measurement (FID #1)
+
+[source, C]
+----
+struct sbiret sbi_measurement_extend(unsigned long measurement_addr,
+                                     unsigned long measurement_index,
+                                     unsigned long hash_algorithm)
+----
+
+Extend the platform TCB measurement with one additional measurement.
+
+This function typically allows the caller to add a software component to the
+list of the TCB layers measurements at any point of the platform lifecycle.
+By supporting this function, the SBI implementation implicitly becomes a
+https://trustedcomputinggroup.org/wp-content/uploads/TCG_D-RTM_Architecture_v1-0_Published_06172013.pdf[Dynamic Root of Trust for Measurement (DRTM)].
+
+Measurements added through this SBI function should be managed separately from
+the other TCB measurements. The SBI implementation can choose to maintain them
+through multiple measurement registers, or extend a single one for every
+`sbi_attestation_extend_measurement` call. In the former case, the caller can
+select a measurement index through the `measurement_index` parameter to extend
+one particular measurement register. When supporting multiple measurement
+registers, SBI implementations should map the supported indexes to the
+https://trustedcomputinggroup.org/wp-content/uploads/TCG_D-RTM_Architecture_v1-0_Published_06172013.pdf[TCG specified DTRM PCR indexes].
+
+All measurements added through this SBI function must be included in the TCB
+attestation evidence generated by the <<get-evidence-function>> function.
+
+The measurement at the `measurement_addr` physical address must be generated
+by a hash function that uses the algorithm specified by `hash_algorithm`.
+`hash_algorithm` must be one of the supported hash algorithms, as reported by
+the `HASH_ALGORITHMS` field from the <<get-capabilities-function>> returned
+structure (`ATST_CAPS`). See
+<<table_attestation_and_measurement_capabilities_format>> for more details.
+
+The possible error codes returned in `sbiret.error` are shown in the
```

Comment:
Nit: Consider active voice..."The <<table_measurement_extend_errors>> table below shows the possible error codes returned in `sbiret.error`"